QgisModelBaker/gui/panel/set_sequence_panel.py

```diff
@@ -29,18 +31,59 @@ class SetSequencePanel(QWidget, WIDGET_UI):
     def __init__(self, parent=None):
         QWidget.__init__(self, parent)
         self.setupUi(self)
+        self.configuration = None
+        self.pg_use_super_login.stateChanged.connect(self._set_superuser)
+        self.last_loaded_sequence_value = None
+        self.sequence_value_edit.setSpecialValueText("-")
 
-    def load_sequence(self, db_connector=None):
+    def set_configuration(self, configuration):
+        self.configuration = configuration
+        self.pg_use_super_login.setVisible(self.configuration.tool & DbIliMode.pg)
+
+    def _set_superuser(self, state):
+        if self.configuration:
+            self.configuration.db_use_super_login = state
+
+    def load_sequence(self):
+        db_connector = db_utils.get_db_connector(self.configuration)
         if db_connector:
-            sequence_value = db_connector.get_ili2db_sequence_value()
-            self.sequence_value_edit.setValue(sequence_value)
-
-    def save_sequence(self, db_connector=None):
-        if self.sequence_group.isChecked():
-            if db_connector:
-                return db_connector.set_ili2db_sequence_value(
-                    self.sequence_value_edit.value()
-                )
-            return False, self.tr("Could not reset T_Id - no db_connector...")
+            self.last_loaded_sequence_value = (
+                db_connector.get_ili2db_sequence_value() or 0
+            )
         else:
-            return True, self.tr("T_Id not set.")
+            self.last_loaded_sequence_value = 0
+
+        # not loaded is 0
+        self.sequence_value_edit.setValue(self.last_loaded_sequence_value)
+        if self.last_loaded_sequence_value == 0:
+            return False, self.tr("Could not load T_Id. Maybe wrong credentials.")
+        else:
+            return True, self.tr("T_Id loaded.")
+
+    def save_sequence(self):
+        if (
+            self.sequence_value_edit.value() != 0
+            and self.sequence_value_edit.value() != self.last_loaded_sequence_value
+        ):
+            # only if it changed
+            if self.sequence_group.isChecked():
+                result, message = False, None
+                db_connector = db_utils.get_db_connector(self.configuration)
+                if db_connector:
+                    result, message = db_connector.set_ili2db_sequence_value(
+                        self.sequence_value_edit.value()
+                    )
+                if not result:
+                    # reset value
+                    if self.last_loaded_sequence_value is None:
+                        self.sequence_value_edit.setSpecialValueText("-")
```

Comment:
```suggestion
                        self.sequence_value_edit.setValue(0)
```
specialvalue text is specified in the constructor

Reply:
or even better `self.sequence_value_edit.setValue(self.sequence_value_edit.minimum())`

On my crusade against literals ⚔️ 😅